src/hdmf/common/__init__.py

```diff
@@ -37,10 +65,14 @@ def register_class(**kwargs):
         def _dec(cls):
             _set_exp(cls)
             __TYPE_MAP.register_container_type(namespace, data_type, cls)
+            cls.type_map = __TYPE_MAP
+            cls.namespace = namespace
             return cls
     else:
         def _dec(cls):
             __TYPE_MAP.register_container_type(namespace, data_type, cls)
+            cls.type_map = __TYPE_MAP
+            cls.namespace = namespace
```

Comment:
I feel uneasy making these public variables that a user can manipulate. Also it seems like this should happen within `register_container_type` instead of having it here and duplicated above.

Reply:
For efficiency yes they could be there. However, having them here gives us:
1. Easy debugging since everything is there.
2. (More importantly) it uses __TYPe_MAP and not a copy. The passing around of copies has proven to be an obstacle for me in that the configuration is not preserved.

Reply:
I do agree I could make a way to avoid the duplication.

Reply:
Please read the answers here on why one should make variables private/protected: https://softwareengineering.stackexchange.com/questions/143736/why-do-we-need-private-variables

> (More importantly) it uses __TYPe_MAP and not a copy. The passing around of copies has proven to be an obstacle for me in that the configuration is not preserved.

This seems like a problem that can be solved in a different way than making a variable that is very intentionally private suddenly very public. When a type map is copied, can the term set config be copied as well?

Reply:
`register_container_type` also already sets the `namespace` attribute here: https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/build/manager.py#L736  (though admittedly 1) that is not very clear and 2) it should probably also not be public)